src/package.cc

```diff
@@ -45,12 +52,16 @@ double Package::GetFillMass(double qty) {
   return fill_mass;
 }
   
-Package::Package() : id_(next_package_id_++), fill_min_(0), fill_max_(std::numeric_limits<double>::max()) {
-  name_ = "unnamed";
-}
-
-Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : name_(name), id_(next_package_id_++), fill_min_(fill_min), fill_max_(fill_max), strategy_(strategy) {}
-
-Package::Package(int id, std::string name) : id_(id), name_(name), fill_min_(0), fill_max_(std::numeric_limits<double>::max()) {}
+Package::Package(std::string name, double fill_min, double fill_max, std::string strategy) : 
```

Comment:
This is my proposal to handle the different cases with a single constructor
1. must always provide at least a name
2. defaults for fill_min, fill_max, and strategy (see header)
3. if the name is unpackaged_name_ set the id manually to the right value, otherwise use the next_id

* This works for creating the unpackaged object
* This works for creating any object without setting anything but the name - you wanted an unnamed package but someone can still do that by specifying "unnamed" in the Create() method.

Reply:
Although, if this is only called through `Create()` then that probably needs defaults if we want to have defaults.